Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -97,13 +161,20 @@ void KrBoxClusterFinder::updateTempCluster(float tempCharge, int tempPad, int te
 std::vector<std::tuple<int, int, int>> KrBoxClusterFinder::findLocalMaxima()
 {
   std::vector<std::tuple<int, int, int>> localMaximaCoords;
-  // loop over whole mMapOfAllDigits the find clusers
-  for (int iTime = 0; iTime < MaxTimes; iTime++) {
-    for (int iRow = 0; iRow < MaxRows; iRow++) {
-      for (int iPad = 0; iPad < MaxPads; iPad++) {
+  // loop over whole mMapOfAllDigits the find clusters
+  for (int iTime = 0; iTime < MaxTimes; iTime++) { //mMapOfAllDigits.size()
+    const auto& mapRow = mMapOfAllDigits[iTime];
+    for (int iRow = 0; iRow < MaxRows; iRow++) { // mapRow.size()
+      // Since pad size is different for each ROC, we take this into account while looking for maxima:
+      setMaxClusterSize(iRow);
 
-        mTempCluster.reset();
-        const float qMax = mMapOfAllDigits[iTime][iRow][iPad];
+      const auto& mapPad = mapRow[iRow];
+      const int padsInRow = mMapperInstance.getNumberOfPadsInRowSector(iRow);
+
+      // Only loop over existing pads:
+      for (int iPad = MaxPads / 2 - padsInRow / 2; iPad < MaxPads / 2 + padsInRow / 2; iPad++) { // mapPad.size()
```

Comment:
This check reduces the time to reconstruct one event from ~17µs to ~13µs. Thanks for pointing it out!